packages/aragon-wrapper/src/index.js

```diff
@@ -228,20 +239,42 @@ export default class Aragon {
     const SET_PERMISSION_EVENT = 'SetPermission'
     const CHANGE_PERMISSION_MANAGER_EVENT = 'ChangePermissionManager'
 
-    const aclObservables = [
-      SET_PERMISSION_EVENT,
-      CHANGE_PERMISSION_MANAGER_EVENT
-    ].map(event =>
-      this.aclProxy.events(event)
-    )
+    const ACL_CACHE_KEY = `acl-${aclAddress}`
+    const REORG_SAFETY_BLOCK_AGE = 100
+
+    // Check if we have cached ACL for this address
+    // Cache object for an ACL: { permissions, blockNumber }
+    const cached = await this.cache.get(ACL_CACHE_KEY, {})
+    const { permissions: cachedPermissions, blockNumber: cachedBlockNumber } = cached
+
+    // When using cache, fetch events from the next block after cache
+    const eventsOptions = cachedPermissions && { fromBlock: (cachedBlockNumber + 1) }
```

Comment:
```suggestion
    const eventsOptions = cachedPermissions ? { fromBlock: (cachedBlockNumber + 1) } : undefined
```

+1 for ternary operator. Thought when `cachedPermissions` is falsey we want to pass in `undefined`, so that the `initializationBlock` is correctly set as the fromBlock.

Reply:
I think both would work, but `undefined` is a little bit faster :).